_posts/2019-02-05-observables.md

```diff
@@ -0,0 +1,168 @@
+	return this._http.get('urlToData' + this.parseFilterToString(filter)).pipe(
+		tap((data) => this.logService.log(data)),
+		map((response) => {
+			let page: Page = new Page();
+			page.data = response.body;
+			page.number = response.headers.get('x-count');
+		})
+	);
+}
+```
+
+In the setup we have so far, both components will use the same service for requesting the data.
+They will both subscribe to an `Observable`, however, it will be a different one.
+When you refresh the data in the datatable, the header will not receive a new value in its subscription and therefore will still show the old number of messages.
+
+In some use cases this might be the desired outcome, but in most cases you want more components to be able to subscribe to the same `Observable`.
+
+# Setup #
+
+In our new setup, we want both, header and datatable, to subscribe to the same Observable, so a call for new data will result in an update in the datatable and the header.
+To make this happen, we will use some kind of layer in between.
+This new layer will provide our components with one and only one and the same `Observable` and will mask the the communication layer from the view's controller.
+Both components will subscribe to this service, so they both get updated by the same result.
+We can do this by creating a simple `Subject` in our service and return that Subject as an Observable to our components.
+We can then implement other calls for this service that will trigger an update of the data, and send it through this subject to both components.
+Because we are not providing a filter when we call the getter for the `Observable` (Subject) we should also find a way of providing the filter to the service, before requerying the data.
+This means we can/are going to use one shared filter, for both components, which makes sense in this case, but not in all use cases.
+
+<div style="text-align: center;" >
+  <img src="/img/observables-the-right-way/second-step.png" width="60%">
+</div>
+
+
+# Refactoring #
+
+We actually don't need to refactor any of our components.
+They will still subscribe to an `Observable` of the service, and react on the incomming data.
+The service however will get refactored.
+Start by defining the Subject as a local property, and because we are going to implement a getter, we can make it private.
+
+data.service.ts
+```javascript
+
+private _data$: Subject<Page> = new BehavioralSubject<>({});
+...
+
+public data(): Observable<Page> {
+	return this._data$ as Observable<Page>;
+}
+
+...
+
+public reload(): void {
+	this._http.get('urlToData' + this.parseFilterToString(this._filter)).pipe(
+		tap((data) => this.logService.log(data)),
+		map((response) => {
+			let page: Page = new Page();
+			page.data = response.body;
+			page.number = response.headers.get('x-count');
+		}).subscribe((page: Page) => {
+			this._data$.next(page);
+		})
+	); // and maybe unsubscribe or throw it way or make a Promise out of it
```

Comment:
```suggestion
	); // and maybe unsubscribe, throw it away or make a Promise out of it
```

Reply:
Removed it... HTTPClient is unsubscribing itself